### ISSUE TITLE
transformer models run in train() mode

### ISSUE THREAD

**Issue by RVirmoors** [RVirmoors/rolypoly#6]
if setting `model->eval()` then outputs remain almost the same for any input

might be something to do with batchnorm / batch sizes?
OR with model initialisation / size?
OR the bug with srk_mask needing to be true/false? TODO check

similar issues & fix ideas:
https://discuss.pytorch.org/t/model-eval-predicts-all-the-same-or-nearly-same-outputs/155025
https://stackoverflow.com/questions/68224016/using-model-eval-on-neural-network-results-in-same-output-everytime-for-very-d
https://discuss.pytorch.org/t/transformer-outputs-nan-in-eval-mode/157936
https://discuss.pytorch.org/t/identical-outputs-for-different-inputs-when-model-in-eval-mode/
